presto-hive/src/test/java/io/prestosql/plugin/hive/TestShowStats.java

```diff
@@ -212,62 +256,145 @@ public void testShowStatsForNonExistingColumnFails()
     }
 
     @Test
-    public void testShowStatsForNonColumnQueryFails()
+    public void testShowStatsForNonColumnQuery()
     {
-        assertQueryFails("SHOW STATS FOR (SELECT 1 FROM nation_partitioned)", ".*Only table columns names are supported in SHOW STATS SELECT clause");
+        assertQuery("SHOW STATS FOR (SELECT 1 FROM nation_partitioned)",
+                "SELECT * FROM (VALUES " +
+                        "   ('_col0', null, 1, 0, null, 1, 1), " +
+                        "   (null, null, null, null, 25, null, null))");
     }
 
     @Test
-    public void testShowStatsForAliasedColumnQueryFails()
+    public void testShowStatsForAliasedColumnQuery()
     {
-        assertQueryFails("SHOW STATS FOR (SELECT nationkey, name, name AS name2 FROM nation_partitioned WHERE regionkey > 0 and regionkey < 4)", ".*Column aliasing is not supported in SHOW STATS SELECT clause");
+        assertQuery("SHOW STATS FOR (SELECT nationkey, name, name AS name2 FROM nation_partitioned WHERE regionkey > 0 and regionkey < 4)",
+                "SELECT * FROM (VALUES " +
+                        "   ('nationkey', null, 5, 0, null, 1, 24), " +
+                        "   ('name', 109, 5, 0, null, null, null), " +
+                        "   ('name2', 109, 5, 0, null, null, null), " +
+                        "   (null, null, null, null, 15, null, null))");
     }
 
     @Test
-    public void testShowStatsForNonIdentifierColumnFails()
+    public void testShowStatsForNonIdentifierColumn()
     {
-        assertQueryFails("SHOW STATS FOR (SELECT CONCAT('some', 'value') FROM nation_partitioned)", ".*Only table columns names are supported in SHOW STATS SELECT clause");
+        assertQuery("SHOW STATS FOR (SELECT CONCAT('some', 'value') FROM nation_partitioned)",
+                "SELECT * FROM (VALUES " +
+                        "   ('_col0', null, 1, 0, null, null, null), " +
+                        "   (null, null, null, null, 25, null, null))");
     }
 
     @Test
-    public void testShowStatsForColumnExpressionFails()
+    public void testShowStatsForColumnExpression()
     {
-        assertQueryFails("SHOW STATS FOR (SELECT nationkey + 1 FROM nation_partitioned)", ".*Only table columns names are supported in SHOW STATS SELECT clause");
+        assertQuery("SHOW STATS FOR (SELECT nationkey + 1 FROM nation_partitioned)",
+                "SELECT * FROM (VALUES " +
+                        "   ('_col0', null, 5, 0, null, 1, 25), " +
```

Comment:
Also, give expression an alias, so that you we do not have to update the tests when implicit column alias changes from _col0 to something else.

There needs to be a test with implicit column names, but let's have one dedicated test case for that.

(same in other places)